sandbox/Sandbox.jsx

```diff
@@ -64,6 +64,11 @@ export default class Sandbox extends React.Component {
         // });
     };
 
+    onDoubleClickNode = (event, id) => {
+        event.stopPropagation(); //??? Marche pas
```

Comment:
I tried running the sandbox in your branch, but I'm getting the following error when I double click on a node:

```
Uncaught TypeError: event.stopPropagation is not a function
```

I debugged it a little and I found out that the *first parameter of the method contains the node id* instead of an event. The `id` parameter is always `undefined`, and no event is ever received by `Sandbox`'s `onDoubleClickNode` function.  

Although you added the `event` parameter in this function, you didn't add it in the `Node` and `Graph` components. The `handleOnClickNode` function in the `Node` component is the one that actually catches the event, and it must pass it to the parent components so it reaches the `Sandbox` callback. To begin with, we must pass the event to the `Graph` component like this:

```
handleOnClickNode = event => this.props.onClickNode && this.props.onClickNode(event, this.props.id);
```

After doing this, you have to also update the `onClickNode` function inside of the `Graph` component to receive the new `event` parameter and then update the call to `this.props.onDoubleClickNode` by passing it the `event` as the first parameter.

```
this.props.onDoubleClickNode(event, clickedNodeId)
```

Once when this is done you could actually try to call event methods from the `Sandbox` component. 🙂

Reply:
This line was supposed to be deleted sorry, it is when I tried to cancel the zoom due to d3-zoom but it also led me to a dead-end, I couldn't cancel it